src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -109,7 +92,10 @@ public class ExtensionsRunner {
      * This field is initialized by a call from {@link ExtensionsInitRequestHandler}.
      */
     public final Settings settings;
-    private final TransportInterceptor NOOP_TRANSPORT_INTERCEPTOR = new TransportInterceptor() {
+    /**
+     * This field is initialized by a call from {@link GetNetty4Transport}.
+     */
+    public final TransportInterceptor NOOP_TRANSPORT_INTERCEPTOR = new TransportInterceptor() {
```

Comment:
You can move this to the newly created `Netty4Transport`class and make it `private`.